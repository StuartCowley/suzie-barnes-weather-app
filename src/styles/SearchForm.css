#searchformbutton {
  font-weight: bold;
  font-size: 20px;
}

.search-form input {
  height: 20px;
  width: 500px;
  font-size: 20px;
}
