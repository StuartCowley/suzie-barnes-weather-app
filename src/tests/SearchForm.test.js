import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../components/SearchForm";

describe("Search Form", () => {
  const validProps = {
    forecasts: [
      {
        date: 1111111,
        description: "Stub description 1",
        icon: 800,
        temperature: {
          max: 22,
          min: 12,
        },
      },
      {
        date: 2222222,
        description: "Stub description 2",
        icon: 602,
        temperature: {
          max: 24,
          min: 13,
        },
      },
    ],
    onSubmit: () => {},
  };
  const newValidProps = {
    forecasts: [
      {
        date: 3333333,
        description: "Stub description 3",
        icon: 800,
        temperature: {
          max: 21,
          min: 10,
        },
      },
      {
        date: 4444444,
        description: "Stub description 4",
        icon: 602,
        temperature: {
          max: 26,
          min: 18,
        },
      },
    ],
    onSubmit: () => {},
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct text when searching", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={newValidProps.searchText}
        setSearchText={newValidProps.setSearchText}
        onSubmit={newValidProps.handleCitySearch}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
