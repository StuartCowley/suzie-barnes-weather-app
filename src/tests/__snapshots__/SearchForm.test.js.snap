// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;

exports[`Search Form renders the correct text when searching 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
